datadog/tests/data_source_datadog_service_level_objectives_test.go

```diff
@@ -0,0 +1,108 @@
+package test
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func TestAccDatadogServiceLevelObjectivesDatasource(t *testing.T) {
+	ctx, accProviders := testAccProviders(context.Background(), t)
+	uniq := strings.ReplaceAll(uniqueEntityName(ctx, t), "-", "_")
+	accProvider := testAccProvider(t, accProviders)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    accProviders,
+		CheckDestroy: testAccCheckDatadogServiceLevelObjectiveDestroy(accProvider),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDatasourceServiceLevelObjectivesIdsConfig(uniq),
+				// Because of the `depends_on` in the datasource, the plan cannot be empty.
+				// See https://www.terraform.io/docs/configuration/data-sources.html#data-resource-dependencies
+				ExpectNonEmptyPlan: true,
+				Check:              checkServiceLevelObjectivesDatasourceAttrs(accProvider, uniq),
+			},
+			{
+				Config: testAccDatasourceServiceLevelObjectivesNameFilterConfig(uniq),
+				// Because of the `depends_on` in the datasource, the plan cannot be empty.
+				// See https://www.terraform.io/docs/configuration/data-sources.html#data-resource-dependencies
+				ExpectNonEmptyPlan: true,
+				Check:              checkServiceLevelObjectivesDatasourceAttrs(accProvider, uniq),
+			},
+			{
+				Config: testAccDatasourceServiceLevelObjectivesTagsFilterConfig(uniq),
+				// Because of the `depends_on` in the datasource, the plan cannot be empty.
+				// See https://www.terraform.io/docs/configuration/data-sources.html#data-resource-dependencies
+				ExpectNonEmptyPlan: true,
+				Check:              checkServiceLevelObjectivesDatasourceAttrs(accProvider, uniq),
+			},
+			{
+				Config: testAccDatasourceServiceLevelObjectivesMetricsFilterConfig(uniq),
+				// Because of the `depends_on` in the datasource, the plan cannot be empty.
+				// See https://www.terraform.io/docs/configuration/data-sources.html#data-resource-dependencies
+				ExpectNonEmptyPlan: true,
+				Check:              checkServiceLevelObjectivesDatasourceAttrs(accProvider, uniq),
+			},
+		},
+	})
+}
+
+func checkServiceLevelObjectivesDatasourceAttrs(accProvider *schema.Provider, uniq string) resource.TestCheckFunc {
+	return resource.ComposeTestCheckFunc(
+		resource.TestCheckResourceAttrSet("data.datadog_service_level_objectives.foo", "slos.0.id"),
+		resource.TestCheckResourceAttr("data.datadog_service_level_objectives.foo", "slos.0.name", uniq),
+		resource.TestCheckResourceAttr("data.datadog_service_level_objectives.foo", "slos.0.type", "metric"),
+	)
+}
```

Comment:
I think it'd be helpful here to assert at least 2 SLOs are returned. Would you be able to add a new config for this?